apps/developer-hub/src/components/EntropyTable/index.tsx

```diff
@@ -120,3 +129,80 @@ const getEntropyDeployments = async (
   const url = isMainnet ? FORTUNA_API_URLS.mainnet : FORTUNA_API_URLS.testnet;
   return await fetchEntropyDeployments(url);
 };
+
+function useEntropyFees(
+  chains: Record<string, EntropyDeployment>,
+): Record<string, bigint | undefined> {
+  const [feesByChain, setFeesByChain] = useState<
+    Record<string, bigint | undefined>
+  >({});
+  const clientsRef = useRef<Map<string, PublicClient>>(new Map());
+
+  function getClient(rpc: string): PublicClient {
+    const cached = clientsRef.current.get(rpc);
+    if (cached) return cached;
+    const client = createPublicClient({
+      transport: http(rpc, { timeout: 10_000, retryCount: 1 }),
+    });
+    clientsRef.current.set(rpc, client);
+    return client;
+  }
+
+  useEffect(() => {
+    const entries = Object.entries(chains);
+    if (entries.length === 0) return;
+
+    let isCancelled = false;
+
+    async function loadInBatches() {
+      const batchSize = 5;
+      for (let i = 0; i < entries.length; i += batchSize) {
+        if (isCancelled) return;
+        const batch = entries.slice(i, i + batchSize);
+        const results = await Promise.allSettled(
+          batch.map(async ([name, d]) => {
+            if (!d.rpc) return [name, undefined] as const;
+            try {
+              const client = getClient(d.rpc);
+              const fee = await client.readContract({
+                address: d.address as `0x${string}`,
```

Comment:
Is this typecasting necessary?  It's always best to avoid typecasting; if the underlying types don't already guarantee the string format matches this then you should use a [type predicate](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) to validate (and handle the case where it doesn't match) rather than typecasting.